apps/web/src/hooks/useMetatags.ts

```diff
@@ -0,0 +1,47 @@
+import { env } from '@latitude-data/env'
+import type { Metadata, ResolvedMetadata } from 'next'
+
+const DEFAULT_TITLE = 'The Open-Source LLM Development Platform'
+const DEFAULT_DESCRIPTION =
+  'Latitude is an end-to-end platform for prompt engineering where domain experts can collaborate with engineers to ship and maintain production-grade LLM features.'
+
+// This hook is necessary to define default metadata correctly, because
+// Next metadata merging would overwrite the nested objects totally.
+export function useMetatags({
```

Comment:
Why not doing in the nextjs server?
https://nextjs.org/docs/app/building-your-application/optimizing/metadata#dynamic-metadata

Reply:
I don't catch what do you exactly mean. The metadata is always generated in the server. I am already using generateMetadata wherever I need to generate dynamic metadata that is dependent of the path. For example, a title with name of a project...

I created useMetatags to have sane defaults and to allow correct merging of fields. This is actually advised in that same docs: https://nextjs.org/docs/app/building-your-application/optimizing/metadata#merging

useMetatags does not replace generateMetadata, because it is also used when exporting metadata directly (because the title is static).

Reply:
The `use` name is usually reserved to hooks, which cannot be used in server components. I think that's why @andresgutgon thought this was being used in client components.

Since this is not a hook, as it does not return any reactive data and is ran in server components, I think we should move this somewhere else to avoid confusion

Reply:
Exactly, talked offline

Reply:
Got it, thanks 👍